x/ethbridge/keeper/keeper.go

```diff
@@ -75,7 +75,10 @@ func (k Keeper) ProcessSuccessfulClaim(ctx sdk.Context, claim string, sugaredLog
 		err = k.supplyKeeper.MintCoins(ctx, types.ModuleName, coins)
 	case types.BurnText:
 		coins = sdk.Coins{sdk.NewCoin(oracleClaim.Symbol, oracleClaim.Amount)}
-		err = k.supplyKeeper.MintCoins(ctx, types.ModuleName, coins)
+		// not mint if burn token is rowan
+		if oracleClaim.Symbol != types.NativeSymbol {
+			err = k.supplyKeeper.MintCoins(ctx, types.ModuleName, coins)
+		}
```

Comment:
Shouldn't we also put logic here to handle the unlocking for rowan.

Reply:
If put the unlocking code in the burn case, then we need copy/paste the err handling and SendCoinsFromModuleToAccount code in different cases. it is not good for code reuse. it is better just handle the token symbol and coin mint according to different case. then use the unique code to handle error and SendCoinsFromModuleToAccount logic

Reply:
I'm unsure why this code is both minting coins, and calling SendCoinsFromModuleToAccount later on line 92.

Reply:
will put the mint out of case branch.